omnigibson/systems/micro_particle_system.py

```diff
@@ -1116,11 +1222,15 @@ def _sync_particle_instancers(cls, idns, particle_groups, particle_counts):
             idn = cls.particle_instancer_name_to_idn(name=name)
             info = idn_to_info_mapping[idn]
             instancer = cls.particle_instancers[name]
-            if instancer.particle_group != info["group"] or instancer.n_particles != info["count"]:
-                logging.debug(f"Got mismatch in particle instancer {name} when syncing, deleting and recreating instancer now.")
-                # Add this instancer to both the delete and creation pile
-                instancers_to_delete.add(name)
-                instancers_to_create.add(name)
+            if instancer.particle_group != info["group"]:
+                instancer.particle_group = info["group"]
+            count_diff = info["count"] - instancer.n_particles
+            if count_diff > 0:
+                # We need to add more particles to this group
+                instancer.add_particles(positions=np.zeros((count_diff, 3)))
```

Comment:
Who is responsible for setting these particles to the correct position? load_state?